devtools/gradle/build.gradle

```diff
@@ -83,3 +94,29 @@ gradlePlugin {
 wrapper {
     distributionType = Wrapper.DistributionType.ALL
 }
+
+/*
+ * Functional tests for this module.
+ */
+task functionalTest(type: Test, description: 'Functional tests', dependsOn: [processResources, classes]) {
+    description = 'Run the functional tests.'
+    group = 'verification'
+
+    useJUnitPlatform()
+    testClassesDirs = sourceSets.funcTest.output.classesDirs
+    classpath = sourceSets.funcTest.runtimeClasspath
+    environment 'QUARKUS_VERSION', "${version}"
+    mustRunAfter test
+
+    // propagate the custom local maven repo, in case it's configured
+    if (System.properties.containsKey('maven.repo.local')) {
+        systemProperty 'maven.repo.local', System.properties.get('maven.repo.local')
+    }
+    testLogging {
+        events "passed", "skipped", "failed"
+    }
+    // Circumvent issue: https://github.com/gradle/gradle/issues/1675
+    systemProperty 'org.gradle.testkit.dir', file("${buildDir}/tmp/test-kit")
+}
+
+check.dependsOn functionalTest
```

Comment:
The `build` task depends on the `check` task. So, `mvn clean install` will in-turn trigger the functional tests